pkg/didcomm/protocol/introduce/states.go

```diff
@@ -0,0 +1,187 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package introduce
+
+import (
+	"errors"
+
+	"github.com/hyperledger/aries-framework-go/pkg/didcomm/dispatcher"
+)
+
+const (
+	// common states
+	stateNameNoop  = "noop"
+	stateNameStart = "start"
+	stateNameDone  = "done"
+
+	// introducer states
+	stateNameArranging  = "arranging"
+	stateNameDelivering = "delivering"
+	stateNameConfirming = "confirming"
+	stateNameAbandoning = "abandoning"
+
+	// introducee states
+	stateNameDeciding = "deciding"
+	stateNameWaiting  = "waiting"
+)
+
+// The introduce protocol's state.
+type state interface {
+	// Name of this state.
+	Name() string
+	// Whether this state allows transitioning into the next state.
+	CanTransitionTo(next state) bool
+	// Executes this state, returning a followup state to be immediately executed as well.
+	// The 'noOp' state should be returned if the state has no followup.
+	Execute(msg dispatcher.DIDCommMsg) (followup state, err error)
+}
+
+type noOp struct {
+}
+
+func (s *noOp) Name() string {
+	return stateNameNoop
+}
+
+func (s *noOp) CanTransitionTo(_ state) bool {
+	return false
+}
+
+func (s *noOp) Execute(_ dispatcher.DIDCommMsg) (state, error) {
+	return nil, errors.New("cannot execute no-op")
+}
+
+// start state
+type start struct {
+}
+
+func (s *start) Name() string {
+	return stateNameStart
+}
+
+func (s *start) CanTransitionTo(next state) bool {
+	// Introducer can go to arranging or done state
+	// TODO: need clarification: Introducee can go to stateNameDeciding
```

Comment:
This is correct. There is no TODO required here, or is there?

Reply:
I added todo bc in State Machine it was `transition to "invited"` I suppose it was a mistake it should be `transition to "deciding"` instead of `transition to "invited"`.

Reply:
We should just remove the TODO and let #288 sort it out.